### DISCUSSION TITLE
Click on mouse button up, not mouse button down?

### DISCUSSION THREAD

**Discussion by HubKing** [tomasklaen/uosc#545] [Q&A] [answered]
The default UI's click is registered when the mouse button gets released, not when it is pressed. Other applications work that way, too, whether it's Windows or Gnome. But it seems that UOSC registers a click on mouse button down. Is there any reason why it acts differently? Can this be changed to mouse button up?

**Comment by christoph-heinrich** [comment, marked as answer]
> Can this be changed to mouse button up?

Wouldn't be hard to change, but I never saw a problem with how it is now.
Elements have `handle_cursor_down()` and `handle_cursor_up()` functions, in most cases it should be fine to just replace down with up. Some extra care needs to be taken with elements that can be dragged (e.g. menu).

**Comment by tomasklaen** [comment]
Because it's easier to implement and feels more responsive. Up event is currently reserved for clearing drags. Changing clicks to up would require refactoring cursor event handling and adding complexity to the code, and I don't see a good reason for doing that.

**Reply by HubKing** [reply]
Well, I have been often using releasing mouse at somewhere else (moving away from the mouse-down position) as a way to cancel a click, or at least I can have some time to think before releasing the mouse click. It seems bizarre to me that MPV gets closed on mouse down on the [X] button, because no other apps work that way, but if all other users think it is fine, than maybe there is no need to change that.
